extensions/graph/pkg/identity/ldap.go

```diff
@@ -352,13 +355,32 @@ func (i *LDAP) getLDAPUserByFilter(filter string) (*ldap.Entry, error) {
 	return res.Entries[0], nil
 }
 
-func (i *LDAP) GetUser(ctx context.Context, nameOrID string) (*libregraph.User, error) {
+func (i *LDAP) GetUser(ctx context.Context, nameOrID string, queryParam url.Values) (*libregraph.User, error) {
 	i.logger.Debug().Str("backend", "ldap").Msg("GetUser")
 	e, err := i.getLDAPUserByNameOrID(nameOrID)
 	if err != nil {
 		return nil, err
 	}
-	return i.createUserModelFromLDAP(e), nil
+	userGroups, err := i.getGroupsForUser(e.DN)
```

Comment:
This query should go into the `if slices.Contains(sel, "memberOf") || slices.Contains(exp, "memberOf")` block. We should only do it if group memberships are explicitly requested.